c++/src/kj/compat/tls.c++

```diff
@@ -443,8 +532,39 @@ public:
   }
 
 private:
+  Promise<void> doOneAccept() {
+    // Accept one connection and queue up the next accept on our TaskSet
+
+    auto loopAgainOnSuccess = [this](AuthenticatedStream&& stream) {
+      auto promise = kj::evalNow([&] {
+        // Do the SSL acceptance procedure.
+        return tls.wrapServer(kj::mv(stream));
+      });
+
+      // queue.push() fowards any exceptions to its returned promise, so our error handler will
+      // catch if promise fails.
+      tasks.add(queue.push(kj::mv(promise)));
+
+      // Queue up the next accept loop immediately without waiting for SSL_accept()/wrapServer().
+      tasks.add(doOneAccept());
+    };
+
+    auto loopAgainOnFailure = [this](Exception&& e) {
+      KJ_LOG(ERROR, "TlsConnectionReceiver could not accept()", kj::mv(e));
+
+      // Even though we had a bad accept(), we should still loop again.
+      tasks.add(doOneAccept());
+    };
+
+    return inner->acceptAuthenticated().then(
+        kj::mv(loopAgainOnSuccess), kj::mv(loopAgainOnFailure));
```

Comment:
Oh man, you're going to have a fun time reading the codebase. :)

Reply:
Yeah, probably! I'll move towards improving readability where I can and put up with it where I can't.

Reply:
Of course, people can reasonably disagree on what style is most readable. Consistency is important since people will always have an easier time reading patterns they are used to. It's definitely frustrating getting into a new codebase that uses different style, because everything seems less readable at first, since it's not what you're used to... but that doesn't necessarily mean the style is objectively less readable.

Personally I find it more difficult to follow code with the lambdas broken out like you have here, because it means the control flow bounces around more, and I have to cross-reference the names with the definitions. I'm used to being able to just read top-to-bottom and see everything in roughly the order it happens. Not saying that what I'm used to is better or worse than what you prefer, but it is what we do throughout the KJ and Workers codebases.

Reply:
I understand that a linear vertical flow is helpful for understanding as well! Keeping in mind that this is async code, perhaps would be most readible to both of us as something like this:
```
auto promise = ...
auto onSuccess = ...
auto onFailure = ...
return promise.then(onSuccess, onFailure);
```

I rely a lot on horizontal indent to convey scoping. When the indentation shifts left for a block, that tells me that I've left a scope, which isn't true here. There's a lot in the codebase that's different than what I'm used to, but this is the one thing that feels like it decreases my ability to understand the code quickly even when I'm familiar with the style.

Reply:
Thinking about this a bit more, I'd like to make it clear that I'm not trying to be trouble here. I have a vested interest in finding a style of coding that fits within our rules and is readable to us both. I'm curious if it's worth adding a line to the guide on preferring inline lambdas if you feel out of line actively thwarts comprehension.